src/components/AllComponents.js

```diff
@@ -0,0 +1,539 @@
+import React, { useEffect, useState } from "react";
+import SingleCard from "./SingleCard";
+import SuggestToWeb from "./SuggestToWeb";
+import { handleRarityChange } from "../util/helpers";
+
+import { useQuery, useLazyQuery } from "@apollo/client";
+import { QUERY_CHARACTERS, QUERY_ONECHARACTER, GET_USERDATA, GET_USERCHARACTERSBYID } from "../util/queries";
+import CardDetails from "./CardDetails";
+
+import Auth from "../util/auth";
+
+function AllComponents() {
+  const categorySelect = document.getElementById("categories");
+
+  const [search, setSearch] = useState("");
+  const [characters, setCharacters] = useState([]);
+  const [characterCategory, setCharacterCategory] = useState("");
+
+  const [rarityCategory, setRarityCategory] = useState("ALL");
+  const [filteredCharacters, setFilteredCharacters] = useState([]);
+  const [typeCharacters, setTypeCharacters] = useState("ALL");
+  const [cardDetails, setCardDetails] = useState({
```

Comment:
This is fairly brittle, especially since some data is scraped from another site. Are you not able to set the initial card from the first item in allCharacters once the GQL data loads, and until then initialize to empty/null?